### PR TITLE
Fixes #12618 - Now calls for compute attributes are explicit.

### PR DESCRIPTION
**Pull request by ShimShtein**
Replaces https://github.com/theforeman/foreman/pull/2937.

### CODE REVIEW

**Comment by dLobatog** [comment]
[test] because Katello job is unstable

File: app/models/host/managed.rb [hunk 1]
```diff
@@ -604,9 +604,7 @@ def hash_clone(value)
 
   def set_hostgroup_defaults
     return unless hostgroup
-    assign_hostgroup_attributes(%w{domain_id compute_profile_id})
```
Comment (domcleal) [inline app/models/host/managed.rb:607]: Does this prevent inheritance of the compute_profile_id in the case of an API request?  Should it still be inherited when possible, but we just skip the set_compute_attrs step?

**Comment by domcleal** [comment]
There should also be a test covering the change to the controller action, there's only a unit test of the method it's calling.

File: app/models/host/managed.rb (hunk 1 above)
Comment (ShimShtein) [inline app/models/host/managed.rb:607]: It is handled by `#apply_inherited_attributes`. Look at: https://github.com/theforeman/foreman/blob/develop/app/models/host/managed.rb#L568

**Comment by ShimShtein** [comment]
functional test added

File: app/controllers/hosts_controller.rb [hunk 2]
```diff
@@ -560,6 +560,7 @@ def process_hostgroup
               Host.new(params[:host])
             end
     @host.set_hostgroup_defaults
+    @host.set_compute_attributes
```
Comment (domcleal) [inline app/controllers/hosts_controller.rb]: Should this only be called if no compute_profile_id is passed via params (i.e. it's inherited)?  Changing the hostgroup when not inheriting the compute profile currently clears the attributes.

File: test/functional/hosts_controller_test.rb [hunk 3]
```diff
@@ -963,6 +963,35 @@ def test_submit_multiple_rebuild_config_pessimistic
...
+  test '#process_hostgroup changes compute attributes' do
+    Fog.mock!
+    group1 = FactoryGirl.create(:hostgroup, :compute_profile => compute_profiles(:one))
+    host = FactoryGirl.create(:host, :managed, :on_compute_resource)
+    host.stubs(:queue_compute)
+    host.hostgroup = group1
+    host.compute_resource = compute_resources(:one)
+    host.compute_profile = compute_profiles(:one)
+    host.set_compute_attributes
+    host.save!
```
Comment (domcleal) [inline test/functional/hosts_controller_test.rb]: Is the `save!` (and `.create` earlier) required?  A new/unsaved host should be equivalent to the New Host form.

**Comment by ShimShtein** [comment]
Removed unnecessary `save!` leftovers.

**Comment by domcleal** [comment]
There's also an inline comment here: https://github.com/theforeman/foreman/pull/2957#discussion_r47067462 - could you address/reply to it please?

File: app/controllers/hosts_controller.rb (hunk 2 above)
Comment (ShimShtein) [inline app/controllers/hosts_controller.rb]: No, If I change the compute_profile manually from UI, I still want to reset the attributes.

Comment (domcleal) [inline app/controllers/hosts_controller.rb]: Doesn't that call the compute resource controller to refresh the VM tab, rather than process_hostgroup?

**Comment by ShimShtein** [comment]
Now compute attributes will change only when compute_profile will be set to "inherit"

**Comment by domcleal** [comment]
Merged as 19d9e22, thanks @ShimShtein!

**Pull request closed without merging**